contracts/GoodGhostingPolygonCurve.sol

```diff
@@ -0,0 +1,674 @@
+    uint256 public immutable earlyWithdrawalFee;
+    /// @notice The performance admin fee (percentage)
+    uint256 public immutable customFee;
+    /// @notice Defines the max quantity of players allowed in the game
+    uint256 public immutable maxPlayersCount;
+    uint256 public immutable poolType;
+    /// @notice winner counter to track no of winners
+    uint256 public winnerCount = 0;
+    /// @notice total tokens in aave pool
+    uint256 public constant numAaveTokens = 3;
+    /// @notice total tokens in atricrypto pool
+    uint256 public constant numAtricryptoTokens = 5;
+    /// for some reason the curve contracts have int128 as param in the withdraw function
+    /// hence the two types since type conversion is not possible
+    /// @notice token index in the pool in int form
+    int128 inboundTokenIndexInt;
+    /// @notice token index in the pool in uint form
+    uint256 inboundTokenIndexUint;
+
+    /// @notice controls if admin withdrew or not the performance fee.
+    bool public adminWithdraw;
+    /// @notice Controls if tokens were redeemed or not from the pool
+    bool public redeemed;
+    /// @notice pool address
+    ICurvePool public pool;
+    /// @notice curve lp token
+    IERC20 public lpToken;
+    /// @notice Address of the token used for depositing into the game by players (DAI)
+    IERC20 public immutable daiToken;
+    /// @notice Defines an optional token address used to provide additional incentives to users. Accepts "0x0" adresses when no incentive token exists.
+    IERC20 public immutable incentiveToken;
+    /// @notice gauge address
+    ICurveGauge public immutable gauge;
+    /// @notice curve token
+    IERC20 public immutable curve;
+    /// @notice wmatic token
+    IERC20 public immutable matic;
+
+    struct Player {
+        bool withdrawn;
+        bool canRejoin;
+        bool isWinner;
+        address addr;
+        uint256 mostRecentSegmentPaid;
+        uint256 amountPaid;
+        uint256 winnerIndex;
+    }
+    /// @notice Stores info about the players in the game
+    mapping(address => Player) public players;
+    /// @notice controls the amount deposited in each segment that was not yet transferred to the external underlying pool
+    /// @notice list of players
+    address[] public iterablePlayers;
+    /// @notice list of winners
+    address[] public winners;
+
+    event JoinedGame(address indexed player, uint256 amount);
+    event Deposit(
+        address indexed player,
+        uint256 indexed segment,
+        uint256 amount
+    );
+    event Withdrawal(
+        address indexed player,
+        uint256 amount,
+        uint256 playerReward,
+        uint256 playerCurveReward,
+        uint256 playerIncentive
+    );
+    event FundsRedeemedFromExternalPool(
+        uint256 totalAmount,
+        uint256 totalGamePrincipal,
+        uint256 totalGameInterest,
+        uint256 rewards,
+        uint256 curveRewards,
+        uint256 totalIncentiveAmount
+    );
```

Comment:
nothing wrong here, just interested to learn how you're gonna handle different events in the subgraphs, specially having different orders in the fields (rather than just additional fields from the current aave version to the curve version)

Reply:
it would be a diff. subgraph branch altogether with a different abi